### PR TITLE
Fix/DAO Upgrade Proposal Enhacemenets

### PR DESCRIPTION
**Pull request by roshan-bisht-lab49**
**Optimised the parsing method for DAO upgrade proposals**

### CODE REVIEW

File: src/dao/hooks/useGovernanceDAOProposals.tsx [hunk 1]
```diff
@@ -91,7 +91,6 @@ export function useGovernanceDAOProposals(
           proxyAdmin: proposalData?.proxyAdmin ?? "",
           proxyLogic: proposalData?.proxyLogic ?? "",
           currentLogic: proposalData?.currentLogic ?? "",
-          isAdminApproved: proposalData?.isAdminApproved ?? false,
           isAdminApprovalButtonVisible: proposalData?.isAdminApprovalButtonVisible ?? false,
```
Comment (damiano-melcarne-lab49) [inline src/dao/hooks/useGovernanceDAOProposals.tsx:94]: UI state, such as visibility, should be controlled at the component level. Application level `react-query` hooks should be used to fetch/cache/return server state.

File: src/dao/hooks/useDAOProposals.tsx [hunk 2]
```diff
@@ -81,6 +89,28 @@ export function useDAOProposals(
     }
   }
 
+  async function getUpgradeProposalData(proposalData: DAOUpgradeProposal) {
```
Comment (damiano-melcarne-lab49) [inline src/dao/hooks/useDAOProposals.tsx:92]: Move this function to the service layer (DAO service). This is essentially an API endpoint used for fetching Contract Upgrade Proposal data.

File: src/dao/hooks/useDAOProposals.tsx [hunk 3]
```diff
@@ -81,6 +89,28 @@ export function useDAOProposals(
...
+    const changeAdminLogs = allEvents.get(DAOEvents.ChangeAdmin) ?? [];
+    const upgradedLogs = allEvents.get(DAOEvents.Upgraded) ?? [];
+
+    const currentLogic = isNotNil(upgradedLogs[0])
+      ? (await DexService.fetchContractId(upgradedLogs[0].implementation)).toString()
+      : "";
+    const latestAdminLog = isNotNil(changeAdminLogs[0]) ? changeAdminLogs[0] : "";
+    const proxyAdmin = AccountId.fromSolidityAddress(proposalData.proxyAdmin).toString();
+    const proxyLogic = (await DexService.fetchContractId(proposalData.proxyLogic)).toString();
+    const parsedProposalDataData = { ...proposalData, proxyAdmin, proxyLogic, currentLogic };
```
Comment (damiano-melcarne-lab49) [inline src/dao/hooks/useDAOProposals.tsx:106]: Can be shortened to `parsedProposalData`.

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]

**Comment by damiano-melcarne-lab49** [comment]
@roshan-bisht-lab49 Can we close this PR?

**Comment by roshan-bisht-lab49** [comment]
> @roshan-bisht-lab49 Can we close this PR?

Yeah I am closing it, I will do it more refactoring.

**Pull request closed without merging**